pkgs/puppy/lib/src/map_command.dart

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2025, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:args/command_runner.dart';
+import 'package:build_cli_annotations/build_cli_annotations.dart';
+import 'package:io/ansi.dart';
+
+import 'constants.dart';
+
+part 'map_command.g.dart';
+
+class MapCommand extends _$MapArgsCommand<void> {
+  @override
+  String get description =>
+      'Run the provided command in each subdirectory containing '
+      '`pubspec.yaml`.';
+
+  @override
+  String get name => 'map';
+
+  @override
+  Future<void>? run() async {
+    await _doMap(_options);
+  }
+}
+
+@CliOptions(createCommand: true)
```

Comment:
Is there a way to pipe through command aliases here? If we wanted to instead support `puppy run`?

Reply:
Oh crap! I forgot! Just submit a PR to change that now.